src/write.c

```diff
@@ -213,20 +281,37 @@ void avifEncoderSetCodecSpecificOption(avifEncoder * encoder, const char * key,
     avifCodecSpecificOptionsSet(encoder->csOptions, key, value);
 }
 
-static void avifEncoderWriteColorProperties(avifRWStream * s, const avifImage * imageMetadata, struct ipmaArray * ipma, uint8_t * itemPropertyIndex)
+static void avifEncoderWriteColorProperties(avifRWStream * outputStream,
+                                            const avifImage * imageMetadata,
+                                            struct ipmaArray * ipma,
+                                            avifItemPropertyDedup * dedup)
 {
+    avifRWStream * s = outputStream;
+    if (dedup) {
```

Comment:
These `if (dedup)` checks should be omitted because we never call this function with `dedup = NULL`.

If the  `if (dedup)` checks were added because of static analysis warnings, we can deal with that by adding `assert(dedup)` just like the assert(ipma) at line 291.

Reply:
`avifEncoderWriteColorProperties()` is usage when writing tracks, which doesn't leverage `ipma`. Therefore, it is legal to call this function with `NULL` for the last two arguments, and this function must honor that. This is why `if (dedup)` is sprinkled throughout the function. We could make `avifItemPropertyDedupStart()` internally `NULL` check and early-out to save on some code paperwork in this function, but I think it would cost the function a bit of readability of intent. I'll add a comment above this function making it clear that it is legal to pass `NULL` for these arguments.

Reply:
I thought there was only one call to avifEncoderWriteColorProperties(). Sorry about my confusion.